src/util/parse_options.cpp

```diff
@@ -76,6 +76,11 @@ int parse_options_baset::main()
     std::cerr << e.what() << "\n";
     return CPROVER_EXIT_USAGE_ERROR;
   }
+  catch(const incorrect_goto_program_exceptiont &e)
```

Comment:
Isn't that subsumed by the case below?

Reply:
Yes, I guess the intention is to distinguish individual exception classes, maybe different error message or return code. There is not dedicated error code for incorrect goto programs, only for `PARSE_ERROR`.

Reply:
parse_options_baset is mean to be very broadly applicable -- including in programs that do not know anything about goto_programs. I would go for a broader superclass here.

Reply:
Ok, catching `incorrect_goto_program` is omitted, and `cprover_exception_baset` is caught directly.